shell.nix

```diff
@@ -0,0 +1,17 @@
+{ system ? builtins.currentSystem }:
+let
+  # nixpkgs / devshell is only used for development. Don't add it to the flake.lock.
+  devshellGitRev = "709fe4d04a9101c9d224ad83f73416dce71baf21";
+
+  nixpkgsSrc = ./.;
+
+  devshellSrc = fetchTarball {
```

Comment:
Flakes shouldn't use `builtins.{fetchTarball,fetchGit,...}. This may even become impossible in the future. Dependencies should be expressed at the flake level so they can be queried and updated in a uniform way.

Reply:
Thanks for pointing this out! We had a large discussion over at [`divnix/devos`](https://github.com/divnix/devos) if this is an antipattern or actually a good pattern. Most lock files, support some notion of development dependencies and "usage" dependency (for not to say runtime).

For example [`flake-utils-plus`](https://github.com/gytis-ivaskevicius/flake-utils-plus) like `numtide/flake-utils` has a policy to depend on nothing, but builtins. This makes a lot of sense so that no subtle incompatibilities would be introduced since coded against one nixpkgs version but instantiated with another.

For those packages it would be absolutely out of question to provide such development amentities. Until flakes support dev-only (optional) dependencies, which are not required for using a flake, this is the only pattern we could find to map this use case concisely into code.